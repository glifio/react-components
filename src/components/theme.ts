export const baseColors = {
  mono: {
    black: '#000',
    nearblack: '#262626',
    darkgray: '#666666',
    lightgray: '#C4C4C4',
    silver: '#999999',
    white: '#ffffff',
    transparent: 'transparent'
  },
  blue: {
    lightest: '#F6F8FE',
    lighter: '#EFF3FD',
    light: '#E4EBFC',
    mid: '#d1ddfa',
    mid2: '#bacbf7',
    muted: '#e8ecf7',
    muted700: '#d9e0f2'
  },
  green: {
    primary: '#1AD08F',
    light: '#D2F5ED',
    muted: '#85e0cb',
    dark: '#007056',
    darker: '#08442F'
  },
  red: {
    light: '#FC6D6F',
    dark: '#660000',
    base: '#ff4242'
  },
  orange: {
    light: '#FF9E80',
    dark: '#661800',
    mid: '#FFA826'
  },
  yellow: {
    light: '#FFDC99',
    deep: '#FCA703',
    mid: '#F7C103'
  },
  purple: {
    light: '#E0D7FE',
    mid: '#5E26FF',
    deep: '#290a85'
  }
}

// The core color object is the only non-semantic object here. This is because these colors are used so widely that it would be highly redundant to replicate these color values repeatedly within this "colors" object to style the text/background of every single Component the app renders.
const core = {
  primary: baseColors.purple.mid,
  secondary: baseColors.purple.light,
  tertiary: baseColors.purple.deep,
  black: baseColors.mono.black,
  nearblack: baseColors.mono.nearblack,
  darkgray: baseColors.mono.darkgray,
  lightgray: baseColors.mono.lightgray,
  silver: baseColors.mono.silver,
  white: baseColors.mono.white,
  transparent: baseColors.mono.transparent,
  yellow: baseColors.yellow.deep
}

const colors = {
  core,
  buttons: {
    primary: {
      background: baseColors.green.primary,
      borderColor: baseColors.green.primary,
      color: baseColors.green.darker
    },
    secondary: {
      background: baseColors.mono.transparent,
      borderColor: baseColors.mono.nearblack,
      color: baseColors.mono.nearblack
    },
    tertiary: {
      background: baseColors.mono.transparent,
      borderColor: baseColors.purple.light,
      color: baseColors.purple.light
    },
    'tertiary-selected': {
      background: baseColors.purple.mid,
      borderColor: baseColors.purple.mid,
      color: baseColors.purple.light
    }
  },
  background: {
    app: baseColors.blue.lightest,
    screen: baseColors.blue.lighter,
    text: baseColors.blue.light,
    messageHistory: baseColors.blue.muted
  },
  card: {
    account: {
      background: core.primary,
      color: core.secondary
    },
    balance: {
      background: baseColors.mono.transparent,
      color: core.nearblack
    },
    confirmation: {
      background: baseColors.green.light,
      foreground: baseColors.green.dark
    },
    error: {
      background: baseColors.red.light,
      foreground: baseColors.red.dark
    }
  },
  input: {
    background: {
      base: baseColors.blue.mid,
      active: baseColors.blue.mid2,
      valid: baseColors.green.muted,
      invalid: baseColors.red.light
    },
    border: core.silver
  },
  status: {
    success: {
      background: baseColors.green.primary,
      foreground: baseColors.green.darker
    },
    pending: {
      background: baseColors.yellow.deep,
      foreground: baseColors.mono.darkgray
    },
    warning: {
      background: baseColors.orange.light,
      foreground: baseColors.orange.dark
    },
    fail: {
      background: baseColors.red.light,
      foreground: baseColors.red.dark
    },
    inactive: baseColors.mono.lightgray
  },
  // gonna start slowly migrating colors from their inheritance vars above
  blue: {
    muted700: '#d9e0f2'
  },
  gray: {
    light: '#C4C4C4',
    medium: '#949598',
    dark: '#666666'
  },
  green: {
    primary: '#1AD08F',
    light: '#D2F5ED',
    muted: '#85e0cb',
    medium: '#007B39',
    dark: '#007056',
    darker: '#08442F'
  },
  yellow: {
    light: '#FFDC99',
    medium: '#C59900',
    deep: '#FCA703',
    mid: '#F7C103'
  },
  purple: {
    light: '#E0D7FE',
    medium: '#5E26FF',
    deep: '#290a85'
  }
}

const theme = {
  colors,
  fontSizes: [
    '0rem',
    '1rem',
    '1.125rem',
    '1.25rem',
    '1.5rem',
    '2rem',
    '3rem',
    '5rem'
  ],
  fontWeights: [0, 400, 600, 900],
  letterSpacings: [0, 1, 2, 4, 8],
  lineHeights: {
    solid: 1,
    title: 1.2,
    copy: 1.4
  },
  textStyles: {
    header: {
      fontSize: 'calc(48px + (48 - 32) * (100vw - 360px) / (1440 - 360))',
      fontWeight: 400,
      margin: 0,
      lineHeight: 'title',
      fontFamily: 'RT-Alias-Grotesk'
    },
    bigTitle: {
      fontSize: 6,
      fontWeight: 700,
      margin: 0,
      lineHeight: 'solid',
      fontFamily: 'RT-Alias-Medium'
    },
    title: {
      fontSize: 4,
      fontWeight: 400,
      margin: 0,
      lineHeight: 'title',
      fontFamily: 'RT-Alias-Grotesk'
    },
    text: {
      fontSize: 2,
      fontColor: core.nearblack,
      fontWeight: 400,
      lineHeight: 'copy',
      fontFamily: 'RT-Alias-Grotesk'
    },
    label: {
      fontSize: 2,
      fontColor: core.darkgray,
      fontWeight: 400,
      lineHeight: 'solid',
      fontFamily: 'RT-Alias-Grotesk'
    },
    num: {
      m: {
        fontSize: 4,
        fontWeight: 400,
        margin: 0,
        lineHeight: 'title',
        fontFamily: 'RT-Alias-Grotesk'
      },
      l: {
        fontSize: 5,
        fontWeight: 700,
        margin: 0,
        lineHeight: 'title',
        fontFamily: 'RT-Alias-Grotesk'
      },
      xl: {
        fontSize: 6,
        fontWeight: 700,
        margin: 0,
        lineHeight: 'solid',
        fontFamily: 'RT-Alias-Grotesk'
      },
      xxl: {
        fontSize: 7,
        fontWeight: 700,
        margin: 0,
        lineHeight: 'solid',
        fontFamily: 'RT-Alias-Grotesk'
      }
    }
  },
  fonts: {
    AliasMedium:
      '"RT-Alias-Medium", "system-ui", "Segoe UI", "Roboto", Helvetica',
    AliasGrotesk:
      '"RT-Alias-Grotesk", "system-ui", "Segoe UI", "Roboto", Helvetica',
    sansSerif:
      '"system-ui", "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";',
    mono: '"system-ui", "Segoe UI", Roboto Mono, Helvetica, Arial, monospace, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";'
  },
  space: [0, 4, 8, 16, 24, 32, 48, 64, 128, 256],
  sizes: [
    0, 4, 8, 16, 24, 32, 48, 64, 80, 160, 240, 300, 480, 560, 640, 720, 800,
    960, 1024, 1200, 1440
  ],
  radii: ['0', '1px', '4px', '8px', '16px', '32px', '100px'],
  borders: [0, `1px solid`, `4px solid`],
  borderStyles: ['solid'],
  borderWidths: ['0', '1px', '2px', '4px'],
  breakpoints: ['40em', '52em', '64em'],
  shadows: [
    '0',
    '0 0.7px 2.2px -8px rgba(0, 0, 0, 0.18),0 1.7px 5.4px  rgba(0, 0, 0, 0.129),0 3.1px 10.1px rgba(0, 0, 0, 0.107),0 5.6px 18.1px rgba(0, 0, 0, 0.09),0 10.4px 33.8px rgba(0, 0, 0, 0.073),0 25px 81px rgba(0, 0, 0, 0.051)',
    'rgba(0, 0, 0, 0.10) 0px 0.7px 2.2px -8px, rgba(0, 0, 0, 0.04) 0px 1.7px 2.4px, rgba(0, 0, 0, 0.106) 0px 3.1px 8.1px, rgba(0, 0, 0, 0.04) 0px 5.6px 12.1px, rgba(0, 0, 0, 0.045) 0px 4.4px 4.8px, rgba(0, 0, 0, 0.05) 0px 15px 41px'
  ],
  opacity: {
    disabled: 0.4
  },
  zIndices: [0, 9, 99, 999, 9999]
}

export default theme

/* EVERYTHING BELOW HERE IS THEMEV2 */

// safe to change
const deviceWidthPxs = {
  phone: 480,
  tablet: 868,
  desktop: 1124
}

const spaces = {
  phone: {
    sm: '',
    small: '',
    default: '8px',
    md: '8px',
    medium: '8px',
    lg: '20px',
    large: '20px'
  },
  default: {
    sm: '',
    small: '',
    default: '10px',
    md: '10px',
    medium: '10px',
    lg: '20px',
    large: '20px'
  },
  tablet: {
    sm: '',
    small: '',
    default: '10px',
    md: '10px',
    medium: '10px',
    lg: '20px',
    large: '20px'
  },
  desktop: {
    sm: '',
    small: '',
    default: '10px',
    md: '10px',
    medium: '10px',
    lg: '20px',
    large: '20px'
  }
}

const fontSizes = {
  phone: {
    sm: '13px',
    small: '13px',
    default: '16px',
    md: '18px',
    medium: '18px',
    lg: '20px',
    large: '20px'
  },
  default: {
    sm: '13px',
    small: '13px',
    default: '18px',
    md: '26px',
    medium: '26px',
    lg: '28px',
    large: '28px'
  },
  tablet: {
    sm: '13px',
    small: '13px',
    default: '18px',
    md: '26px',
    medium: '26px',
    lg: '28px',
    large: '28px'
  },
  desktop: {
    sm: '13px',
    small: '13px',
    default: '18px',
    md: '26px',
    medium: '26px',
    lg: '28px',
    large: '28px'
  }
}

export const maxWidth = '1450px'

// dont change
export const devices = {
  phone: `${deviceWidthPxs.phone}px`,
  tablet: `${deviceWidthPxs.tablet}px`,
  desktop: `${deviceWidthPxs.desktop}px`
}

// these are
// one of size = sm, default, md, lg
// one of device = phone, tablet, desktop
export function space(size = 'default', device = 'default') {
  return spaces[device][size]
}

export function fontSize(size = 'default', device = 'default') {
  return fontSizes[device][size]
}
