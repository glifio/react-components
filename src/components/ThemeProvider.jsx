import PropTypes from 'prop-types'
import { ThemeProvider as StyledThemeProvider } from 'styled-components'
import { theme as defaultTheme } from './theme'

const ThemeProvider = ({ children, theme, ...rest }) => {
  return (
    <StyledThemeProvider theme={theme} {...rest}>
      {children}
    </StyledThemeProvider>
  )
}

ThemeProvider.propTypes = {
  theme: PropTypes.object,
  children: PropTypes.node
}

ThemeProvider.defaultProps = {
  theme: defaultTheme,
  children: <></>
}

ThemeProvider.displayName = 'ThemeProvider'

export default ThemeProvider
