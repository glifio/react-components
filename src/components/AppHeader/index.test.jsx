import { render, cleanup, screen } from '@testing-library/react'

import ThemeProvider from '../ThemeProvider'
import theme from '../theme'
import { Base } from './index.stories'

describe('AppHeader', () => {
  afterEach(cleanup)
  test('renders the box', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Base {...Base.args} />
      </ThemeProvider>
    )
    expect(screen.getByText('Blog')).toBeInTheDocument()
    expect(screen.getByText('Code')).toBeInTheDocument()
    expect(screen.getByText('Nodes')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
