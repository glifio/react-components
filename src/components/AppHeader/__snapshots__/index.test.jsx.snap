// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader renders the box 1`] = `
.c0 {
  height: auto;
}

.c1 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.75rem;
  padding: 0.5em 0.75em;
  border-radius: 100px;
  font-size: 26px;
  font-size: 18px;
  border: 1px solid black;
  color: black;
  background: transparent;
}

.c1:hover {
  color: #5E26FF;
  border-color: #5E26FF;
}

@media (max-width:480px) {
  .c0 {
    width: 25px;
  }
}

@media (min-width:480px) and (max-width:868px) {
  .c0 {
    width: 30px;
  }
}

@media (max-width:868px) {
  .c0 {
    margin-bottom: 2em;
  }
}

@media (min-width:868px) {
  .c0 {
    width: 40px;
  }
}

@media (max-width:480px) {
  .c1 {
    font-size: 18px;
  }
}

@media (max-width:480px) {
  .c1 {
    font-size: 16px;
  }
}

<header
  style="display: flex; justify-content: space-between; padding-bottom: 50px;"
>
  <a>
    <svg
      class="c0"
      fill="none"
      height="71"
      viewBox="0 0 52 71"
      width="52"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        logo-glif
      </title>
      <path
        d="M10.4352 26.1836C10.4354 26.2474 10.4481 26.3105 10.4727 26.3693C10.4973 26.4281 10.5332 26.4815 10.5785 26.5264C10.6237 26.5714 10.6773 26.6069 10.7363 26.6312C10.7952 26.6554 10.8584 26.6677 10.9222 26.6675H50.5292C50.593 26.6677 50.6561 26.6554 50.7151 26.6312C50.7741 26.6069 50.8277 26.5714 50.8729 26.5264C50.9181 26.4815 50.9541 26.4281 50.9786 26.3693C51.0032 26.3105 51.016 26.2474 51.0162 26.1836V10.919C51.016 10.8552 51.0032 10.7922 50.9786 10.7333C50.9541 10.6745 50.9181 10.6211 50.8729 10.5762C50.8277 10.5313 50.7741 10.4957 50.7151 10.4715C50.6561 10.4473 50.593 10.4349 50.5292 10.4351H21.3573C21.2935 10.4349 21.2304 10.4473 21.1714 10.4715C21.1124 10.4957 21.0588 10.5313 21.0136 10.5762C20.9683 10.6211 20.9324 10.6745 20.9078 10.7333C20.8833 10.7922 20.8705 10.8552 20.8703 10.919V15.7485C20.8705 15.8123 20.8833 15.8753 20.9078 15.9342C20.9324 15.993 20.9683 16.0464 21.0136 16.0913C21.0588 16.1362 21.1124 16.1718 21.1714 16.196C21.2304 16.2202 21.2935 16.2326 21.3573 16.2324H42.9C43.515 16.2324 44.1048 16.4767 44.5397 16.9116C44.9746 17.3465 45.2189 17.9363 45.2189 18.5513C45.2189 19.1663 44.9746 19.7562 44.5397 20.191C44.1048 20.6259 43.515 20.8702 42.9 20.8702H18.5514C17.9364 20.8702 17.3465 20.6259 16.9117 20.191C16.4768 19.7562 16.2325 19.1663 16.2325 18.5513V8.1162C16.2325 7.50119 16.4768 6.91136 16.9117 6.47648C17.3465 6.0416 17.9364 5.79729 18.5514 5.79729H50.5292C50.593 5.79749 50.6561 5.78513 50.7151 5.76093C50.7741 5.73672 50.8277 5.70113 50.8729 5.6562C50.9181 5.61127 50.9541 5.55787 50.9786 5.49905C51.0032 5.44024 51.016 5.37715 51.0162 5.31341V0.483881C51.0156 0.355201 50.964 0.232012 50.8727 0.141311C50.7814 0.0506101 50.6579 -0.000205419 50.5292 6.2415e-07H10.9222C10.7935 -0.000205419 10.67 0.0506101 10.5787 0.141311C10.4874 0.232012 10.4358 0.355201 10.4352 0.483881V26.1836Z"
        fill="black"
      />
      <path
        d="M50.5284 47.6879C50.6572 47.6883 50.7809 47.6376 50.8723 47.5468C50.9638 47.4561 51.0155 47.3328 51.0161 47.204V42.3706C51.0155 42.2418 50.9638 42.1185 50.8723 42.0278C50.7809 41.937 50.6572 41.8863 50.5284 41.8867H0.487747C0.358933 41.8863 0.235219 41.937 0.143772 42.0278C0.0523238 42.1185 0.000616192 42.2418 2.46584e-06 42.3706V47.2001C-0.000201704 47.2642 0.0122749 47.3278 0.0367141 47.387C0.0611532 47.4463 0.0970723 47.5001 0.142404 47.5455C0.187736 47.5908 0.241586 47.6267 0.300854 47.6512C0.360122 47.6756 0.423639 47.6881 0.487747 47.6879H50.5284Z"
        fill="black"
      />
      <path
        d="M50.5284 37.2413C50.5921 37.2416 50.6553 37.2293 50.7143 37.2052C50.7733 37.1811 50.827 37.1456 50.8723 37.1008C50.9176 37.0559 50.9536 37.0025 50.9783 36.9438C51.0029 36.885 51.0158 36.8219 51.0161 36.7582V31.9279C51.0155 31.7991 50.9638 31.6758 50.8723 31.585C50.7809 31.4943 50.6572 31.4436 50.5284 31.444H8.1162C7.50119 31.444 6.91136 31.1997 6.47648 30.7648C6.0416 30.3299 5.79729 29.7401 5.79729 29.1251V0.626417C5.7977 0.497602 5.74697 0.373889 5.65625 0.282441C5.56553 0.190993 5.44222 0.139286 5.31341 0.138672H0.483882C0.355068 0.139286 0.231763 0.190993 0.14104 0.282441C0.0503165 0.373889 -0.000408571 0.497602 2.47885e-06 0.626417V36.7582C3.22392e-05 36.8811 0.0473625 36.9993 0.13218 37.0882L0.143002 37.0998C0.187479 37.1443 0.240271 37.1795 0.298363 37.2035C0.356455 37.2275 0.418707 37.2398 0.481564 37.2397H50.5284V37.2413Z"
        fill="black"
      />
      <path
        d="M3.01713e-05 70.3858C-0.000684024 70.4499 0.0112877 70.5136 0.0352565 70.573C0.0592253 70.6325 0.0947178 70.6866 0.139691 70.7323C0.184664 70.778 0.23823 70.8143 0.297306 70.8393C0.356381 70.8642 0.419799 70.8771 0.48391 70.8774H5.31344C5.44225 70.8768 5.56556 70.8251 5.65628 70.7337C5.747 70.6422 5.79773 70.5185 5.79732 70.3897C5.79732 69.904 5.99024 69.4383 6.33365 69.0949C6.67706 68.7515 7.14283 68.5585 7.62848 68.5585H50.5284C50.6572 68.5589 50.7809 68.5082 50.8724 68.4175C50.9638 68.3268 51.0155 68.2035 51.0161 68.0746V63.2413C51.0155 63.1124 50.9638 62.9891 50.8724 62.8984C50.7809 62.8077 50.6572 62.757 50.5284 62.7574H8.11623C7.50121 62.7574 6.91139 62.5131 6.47651 62.0782C6.04163 61.6433 5.79732 61.0535 5.79732 60.4385C5.79732 59.8235 6.04163 59.2336 6.47651 58.7987C6.91139 58.3639 7.50121 58.1196 8.11623 58.1196H50.5284C50.6572 58.12 50.7809 58.0692 50.8724 57.9785C50.9638 57.8878 51.0155 57.7645 51.0161 57.6357V52.8061C51.0155 52.6773 50.9638 52.554 50.8724 52.4633C50.7809 52.3726 50.6572 52.3219 50.5284 52.3223H0.487775C0.35896 52.3219 0.235247 52.3726 0.143799 52.4633C0.0523515 52.554 0.000643898 52.6773 3.01713e-05 52.8061V70.3858Z"
        fill="black"
      />
    </svg>
  </a>
  <nav
    style="display: flex; align-items: flex-start; align-content: flex-start;"
  >
    <a>
      <button
        class="c1"
      >
        Blog
      </button>
    </a>
    <a>
      <button
        class="c1"
        style="margin-left: 10px;"
      >
        Code
      </button>
    </a>
    <a>
      <button
        class="c1"
        style="margin-left: 10px;"
      >
        Nodes
      </button>
    </a>
  </nav>
</header>
`;
